### ISSUE TITLE
Quen changes

### ISSUE THREAD

**Issue by Lknechtli** [Lknechtli/BetterCombatEvolved#16]
Sign +intensity skill changes and quen functionality:

rank 1/5 absorb 1 hit but the % it absorbs should increase.
rank 3/5 takes 2 hit and each rank increases absorb.
final 5/5 rank can take a 3rd hit\

each 10% sign power = 1 second quen duration.  with 10 second base.  250% spell power = 35 second duration
rank 0 = 50% damage absorb
5 % extra per level
rank 5 = 75% damage absorb

possible other change
base reflect damage off of piercing damage , with max rank of that skill completely mirroring damage taken.

**Comment by BardBlue** [comment]
Sounds interesting. I think the worst part about quen is it basically makes you invulnerable. So if it reduces damage instead of eliminating it, it'll be more of a back-up thing instead of freedom to take a hit. And if it takes three hits, it'll be less spammy, too.

Confused about the reflect change idea... base it off piercing damage? So only piercing damage reflects? And full-damage reflect sounds iffy -- it's already really strong at 25% (which, I suppose would be halved with 50% absorb.) It seems to me that ranking up the absorb talent actually makes the mirror reflect damage worse, if the reflect damage is based on damage Geralt takes. I guess that's okay, and just means you are basically choosing between damage reflect, or damage absorb, or a combination.

Is the damage absorb based on the quen intensity talent? If so, that's a great way to make it worth taking. And is the 3/3 rank that increases the hits taken replacing the knockback/damage/stagger on quen being hit? Suppose that's fine.

**Comment by Lknechtli** [comment]
By piercing damage, I mean damage that hits through the shield. So at best, you're dealing the same amount of damage to your opponent as you are taking from them.

Hadn't actually thought of the 3/3. I'll give that some thought.
edit: two different ideas listed.  Should be out of 5 ranks, not 3

**Comment by BardBlue** [comment]
Oh, so you're only going to do one or the other? That is, the 3 hit thing + increase absorb or just increase absorb? Or are you combining both? A great way to make the +sign intensity skill interesting. Any plans to do something similar with the other +sign intensity skills?

You could also make sign intensity increase the damage absorb amount, but obviously you'd want to make it extremely small so you couldn't get 100% damage absorb with the five points in that talent and 200+% sign intensity. Like, 10% sign intensity = .5% more damage absorb. So 250% sign intensity, which is a lot, would still only get you 12.5% more absorb, for 87.5% absorb with those five talent points (or you could reduce the effect of the talent points to 3-4% per point or something while also adding number of hits taken.) Probably fine if intensity just increases duration though. It's just not very exciting after a certain point, but still better than how it is now.

Just throwing ideas out there. Very excited for this change. I really think current quen is kind of ruining the game for me right now, even with a 6 second duration.

**Comment by Lknechtli** [comment]
see 44abb3b4a20de1cc00c84655c759c90c20e5212f
forgot to note that for reflection, it's based off of damage received through quen, up to 50%. So if you take 1000 damage, with everything maxed, you'll only receive 250 damage, and reflect 125 damage back to the attacker.

**Comment by Lknechtli** [comment]
This change was a huge pain lol. glad it's over with, at least for the time being.

**Comment by Lknechtli** [comment]
as for doing it to other skills, I'm considering it.  This change took me a total of like 8 hours to finish, and I'm not even sure that it's bug free at this point.

**Comment by BardBlue** [comment]
Yeah, well, you did basically completely overhaul several aspects + testing different versions. Not sure if the other skills need overhauls as much. A universal but interesting effect for the other skills would save a lot of time (though balancing would take a while).

Should be well worth the time spent if it plays as well as it sounds.

**Comment by BardBlue** [comment]
Ok, after playing with quen for a little while, I like it quite a bit. Though admittedly I haven't tried it with five points in Sign Intensity for 3 75% damage reduced hits (that sounds really quite good).

I feel like I'm reflecting quite a lot of damage with 3 points in the damage reflect talent  and no points in the +sign intensity skill. I still think it's a bit weird that investing more points in the sign intensity perk in a way makes the reflect damage worse (though I suppose if you add up the damage you take in 3 hits, it probably ends up as more damage in total, but still as a smaller percentage of the damage you take than as if you invested no points in the sign intensity perk.) If the damage reflect was a percentage of the damage quen ABSORBED instead of the damage you take, the two talents would synergize better.

Another thing I find confusing is the bit for the alternate quen:

Heals you for 25% of the damage of that hit, or .2_spellpower % of your max health, *_whichever is less**

Wait, why would it choose whichever is less? This just seems like a really awkward attempt to balance the healing. It encourages you you to up your sign intensity or you're going to heal for next to nothing, but it throttles the efficacy of sign intensity based on the damage of the attack. I guess it makes sense as a limiting factor, but it's taken me a while to understand this, and I think a lot of people will find this confusing and off-putting. I suggest you either rephrase it somehow, or rework it to be more simple and intuitive.

I'm trying figure out at what point sign intensity becomes effectively useless -- it depends on whether the damage of the hit can be greater than your max health or not, but also on the max damage enemies realistically do. At 125% sign intensity and 5000 health, it'll take a 5001+ damage hit before 25% of your max health is less than 25% of the damage of the attack. Realistically, almost nothing does that amount of damage, so the max intensity you need is probably more like 75-100%. While sign intensity would still increase basic quen's length, I don't like how limited the effect of intensity is on alternate quen. Maybe .15% of max health per 1% of sign intensity if you really want to keep the concept as it is. 

I think you'd get much of the same effect if you just made it heal you by 5% of the blocked attack or something as a base and increased that with sign intensity by .2% a point or so, maybe .15% if that's too much. So a 100% (with .2% per point) sign intensity would heal you for 25% of the attack. I don't think that's too much healing, and it's a lot easier to understand.

But maybe i"m just misunderstanding the original concept.

**Comment by Lknechtli** [comment]
Alt quen needs some work, I hacked that part together in about an hour.  Not sure how I want to balance it.  I think I basically want it to be a way to block any damage for a set amount of time.  Adding healing to the mix just complicates things.

Also, I might tone down reflect to 10% of absorbed damage

**Comment by BardBlue** [comment]
I like the healing. To me, the alt quen isn't very useful except for this aspect (though it seems to barely heal me for anything even with decently high sign intensity. Maybe I figured the numbers out wrong?) I just figure, if I have time and stamina to put up a shield, I might as well just dodge unless I'm going to get healing out of it.

For the reflect, you mean 10% of absorbed damage per point, or with 5 points invested? Because if the latter, that sounds like a serious nerf, equivalent to a single point of vanilla reflect if you have no points invested in sign intensity, and barely much more even with five points in sign intensity. 15% total might be okay, which would give you 11.25% damage reflects with 5/5 sign intensity. Might even go up to 20%, for an actual 15% damage reflect.

**Comment by Lknechtli** [comment]
for reflect, another option would be to simply make it act like thorn-type
spells in other games - flat damage, scaled by spellpower, and possibly a
knockback at higher levels.  That would completely get rid of the issue of
monsters having large attack compared to their health.

**Comment by BardBlue** [comment]
Meh, I don't mind reflect as a percentage of the attack damage, as long as its reasonable. Flat reflect damage tends to either be useless or OP. Especially now that you actually take damage even with quen on, letting the reflect be a fraction of the damage is a lot more fair, as long as you get the percentage right. Most creatures don't have a stupidly high ratio of damage to health if you're only using like 10% of the damage.

**Comment by Lknechtli** [comment]
I think for alt quen, we might go with it healing up to a max amount instead of %, based on spellpower (maybe 2 health per %).  That would be enough to prevent instant full heals, and still worth using on weaker groups of enemies when combined with the large knockback radius.

Chicken is considering lowering the stamina cost to 40 and adding a 6 second stamina delay to base quen, because the short duration meant he wasn't using it very much unless he was on a sign build.

**Comment by BardBlue** [comment]
A flat heal rate per % sign intensity might work well enough. I'm not sure what you mean by a 6 second stamina delay or how that effects the duration.

**Comment by Lknechtli** [comment]
Whups, he's also increasing the base duration to something like 14 seconds to be more usable at lower levels
